### ISSUE TITLE
iOS: AppStore upload warning: missing 1024x1024 icon

### ISSUE THREAD

**Issue by Joncom** [Kode/Kha#967]
I have a 1024x1024 icon.png file in my project.

![screen shot 2019-01-21 at 12 49 23 pm](https://user-images.githubusercontent.com/1667179/51455217-2591b600-1d7b-11e9-917f-743eb1233460.png)

However, the `node Kha/make ios` process does seem to propagate that into the Xcode project.

![screen shot 2019-01-21 at 12 52 26 pm](https://user-images.githubusercontent.com/1667179/51455278-6d184200-1d7b-11e9-8ccc-5a19b0176586.png)

And then uploading to the AppStore spits out this warning:

![screen_shot_2019-01-19_at_3 17 11_pm](https://user-images.githubusercontent.com/1667179/51455332-9f29a400-1d7b-11e9-8d50-66807fd3ee9a.png)

Basically says apps cannot be released on the AppStore until this is fixed.

**Comment by RobDangerous** [comment]
Hey, I already showed you where to change that.

**Comment by Joncom** [comment]
In `XCodeExporter.ts` right? However, I don't think we've ever discussed the 1024x1024 issue before. I only just saw this warning for the first time, and changed my icon from 512x512 to 1024x1024 yesterday, yet that didn't resolve the warning.

**Comment by RobDangerous** [comment]
Original icon resolution is irrelevant.

**Comment by Joncom** [comment]
```diff
diff --git a/game-haxe/Kha/Kore/Tools/koremake/out/Exporters/XCodeExporter.js b/game-haxe/Kha/Kore/Tools/koremake/out/Exporters/XCodeEx
porter.js
index afdf48ae..5a4b2407 100644
--- a/game-haxe/Kha/Kore/Tools/koremake/out/Exporters/XCodeExporter.js
+++ b/game-haxe/Kha/Kore/Tools/koremake/out/Exporters/XCodeExporter.js
@@ -153,6 +153,7 @@ class XCodeExporter extends Exporter_1.Exporter {
             icons.push(new IconImage('ipad', 40, 2));
             icons.push(new IconImage('ipad', 76, 1));
             icons.push(new IconImage('ipad', 76, 2));
+            icons.push(new IconImage('ios-marketing', 1024, 1));
         }
         else {
             icons.push(new IconImage('mac', 16, 1));
diff --git a/game-haxe/Kha/Kore/Tools/koremake/src/Exporters/XCodeExporter.ts b/game-haxe/Kha/Kore/Tools/koremake/src/Exporters/XCodeExporter.ts
index 99ed6ec9..37b4a5ea 100644
--- a/game-haxe/Kha/Kore/Tools/koremake/src/Exporters/XCodeExporter.ts
+++ b/game-haxe/Kha/Kore/Tools/koremake/src/Exporters/XCodeExporter.ts
@@ -197,6 +197,7 @@ export class XCodeExporter extends Exporter {
                        icons.push(new IconImage('ipad', 40, 2));
                        icons.push(new IconImage('ipad', 76, 1));
                        icons.push(new IconImage('ipad', 76, 2));
+                       icons.push(new IconImage('ios-marketing', 1024, 1));
                }
                else {
                        icons.push(new IconImage('mac', 16, 1));
```

Might do the trick...
